tests/shardddl/run.sh

```diff
@@ -0,0 +1,1586 @@
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4,'ddd');"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,'eee');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,'fff');"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add column new_col1 int after b;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,'ggg');"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,'hhh');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,'iii',9);"
+    run_sql_tidb_with_retry "select count(1) from ${shardddl}.${tb};" "count(1): 9"
+}
+
+function DM_036() {
+    # currently not support pessimistic
+    # run_case 036 "double-source-pessimistic" \
+    # "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b varchar(10));\"; \
+    #  run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b varchar(10));\"; \
+    #  run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b varchar(10));\";" \
+    # "clean_table" "pessimistic"
+
+    run_case 036 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b varchar(10));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b varchar(10));\";" \
+    "clean_table" "optimistic"
+}
+
+function DM_037_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add new_col1 int default 0;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add new_col1 int default -1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(3,3);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add new_col1 int default -1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(6,6);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(7,7);"
+    if [[ "$1" = "pessimistic" ]]; then
+        check_log_contain_with_retry "is different with" $WORK_DIR/master/log/dm-master.log
+    else
+        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+            "query-status test" \
+            "because schema conflict detected" 1
+    fi
+}
+
+function DM_037() {
+    run_case 037 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 037 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_038_CASE() {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3);"
+
+    sleep 1
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add column col1 datetime default now();"
```

Comment:
we may need to detect it ASAP in optimistic mode later, can you help me add a `TODO` in this case?

Reply:
address it in e4fd8d5